x-pack/plugins/security_solution/server/endpoint/routes/actions/service.ts

```diff
@@ -122,40 +108,64 @@ export const getAuditLogResponse = async ({
     ignore: [404],
   };
   const esClient = context.core.elasticsearch.client.asCurrentUser;
-  let result;
-  const params = getAuditLogESQuery({
-    elasticAgentId,
-    from,
-    size,
-  });
+  let actionsResult;
+  let responsesResult;
 
   try {
-    result = await esClient.search(params, options);
+    actionsResult = await esClient.search(
+      getAuditLogESQuery({
+        index: AGENT_ACTIONS_INDEX,
+        elasticAgentId,
+        from,
+        size,
+      }),
+      options
+    );
+    const actionIds = actionsResult.body.hits.hits.map(
+      (e) => (e._source as EndpointAction).action_id
+    );
+
+    responsesResult = await esClient.search(
+      getAuditLogESQuery({
+        index: AGENT_ACTIONS_RESULTS_INDEX,
+        elasticAgentId,
+        actionIds,
+        from,
```

Comment:
So here I've split the query to make it work. Now that I think about paging again, and that the endpoint responses could be many documents later, I believe we need to search the `.fleet-action-responses` index to the very end for every fetch request from the frontend, and then dedupe the received data (if duplicates happen between fetch requests). What do you think @pzl

Reply:
perhaps overcomplicated. You are fetching (and receiving) some N actions in the first query.

Your second query is to the responses index, asking for (up to) N responses. There likely shouldn't be any paging happening in this second query. You are filtering by IDs directly, so you should only be getting the intended ones.

That said, we now have N+2 items in a page. So it might be wise to request `N/2` page size of actions.

Reply:
Makes sense about keeping it simple and not using helper functions, but we still need paging on the second index to get more than 10 items at a time isn't it?

> That said, we now have N+2 items in a page. So it might be wise to request `N/2` page size of actions.

Not really sure if I understand the N+2 items in a page. @pzl

Reply:
front end sends a request for, let's say 20 items.

first, we send the request for actions. How many actions are we requesting? 20? Then we will get 20 actions back.

Then we ask for responses to those 20 actions. We may get (up to) 20 back. Could be 0, or 5, or 19, or 20, depending on things still pending. But potentially up to 20.

We are now returning 40 things total, for a page size of 20. So maybe requesting 10 actions, and getting (possibly) 10 responses will get us closer to requested page size.

For the second request, I'd say crank the `size` param up to (literally) `1000` and don't consider paging it. You will only get `page_size/2` responses, anyway.

Reply:
basically just paginate over the actions index itself, and then pull in the corresponding responses. The overall ordering across pages may not perfectly hold up (if further actions came before some responses). But this will at least not omit things that occurred (bigger problem).